### PR TITLE
Feature/mr 28/activity put delete

### PR DESCRIPTION
**Pull request by lupomartin**
added data in the put and delete tests

### CODE REVIEW

**Review by Micaela-Rossi — COMMENTED** [review]
Check the first section of the file, because i changed all before mergin following the comments. Check if you use the file i merge is okay your tests and don't broke anything.

File: src/tests/activity.test.js [hunk 1]
```diff
@@ -142,3 +142,32 @@ describe('POST /api/activities', () => {
     expect(response.body.error).toBeTruthy();
   });
 });
+describe('PUT /api/activity', () => {
+  test('should return the 200 status if the activity was successfully updated', async () => {
+    const response = await request(app).put(`/api/activity${activitySeed[0]._id.toString()}`).send(mockActivity);
```
Comment (Micaela-Rossi) [inline src/tests/activity.test.js:147]: You are using another url.... the url must be activities not acitivity

**Review by Micaela-Rossi — CHANGES_REQUESTED** [review]
Must change in the controllers/activity.js 
Line 91 from 200 to 201

Line 1, add /* eslint-disable no-underscore-dangle */ and delete line 42, because it's throwing error in some places because you are using _id more than one time.


Replace line 13 with name: `${activitySeed[0].name}`, for correct my mistake so the test doesn't fail if the seed is modified.

Delete line 36, 41, 43 and 44, because it's useless, also change in line 76 and 80 the ${firstActivityId}  for ${activitySeed[0]._id} . (this replace gonna be the same in functionality but it seems better and also save some lines of code).

Also correct the url from /api/acitivity to api/activities from lines 147, 152, 157, 164 and 169.

After the line 34 add the following const to correct some mistakes, 
const mockSameContent = {
  name: `${activitySeed[2].name}`,
  description: `${activitySeed[2].description}`,
  isActive: `${activitySeed[2].isActive}`,
};

const mockUpdate = {
  name: 'Martin',
  description: 'descriptionnn',
  isActive: false,
};

because this change you must change in the line 152 after the .put --> .put(`/api/activities/${activitySeed[2]._id}`).send(mockSameContent);
**IMPORTANT in the controller doesn't exist any validation about update an acitivity with existing content so this test always gonna fail except you change the controller for add this validation.**

And change in line 147, mockActivity for mockUpdate (Also is missing a '/' before the $).
**(only if you wanna keep this validation and update the controller for this)**

In line 169; must add 
const nonExistentId = '64677fdefc13ae39f1753b99';
 And change the delete url to this:
 const response = await request(app).del(`/api/activities/${nonExistentId}`).send();


In line 157:
add same constExistingId as below.
And change the line with url to: 
const response = await request(app).put(`/api/activities/${nonExistentId}`).send();


In line 162 must be status 200 not 201, also in 165 change it. 201 is for correct update.

In line 146 must be 201 status not 200. Also correct in 148.

I think it's all, check if pass the tests if not talk to me to see if i change something more i forgot.

File: src/tests/activity.test.js [hunk 2]
```diff
@@ -142,3 +141,30 @@ describe('POST /api/activities', () => {
...
+describe('DELETE /api/activities', () => {
+  test('should return status 200 if activity was deleted', async () => {
+    const response = await request(app).delete(`/api/activities/${activitySeed[1]._id.toString()}`);
+    expect(response.status).toBe(200);
+    expect(response.body.error).toBeFalsy();
+  });
+  test('should return 400 status if activity not found', async () => {
+    const nonExistentId = '64677fdefc13ae39f1753b99';
+    const response = await request(app).del(`/api/activities/${nonExistentId}`).send();
+
```
Comment (Micaela-Rossi) [inline src/tests/activity.test.js:166]: This line must not be here, after that, all okay!

**Review by Micaela-Rossi — APPROVED** [review]

**Review by nailaghigo — APPROVED** [review]
💯

**Review by nailaghigo — APPROVED** [review]
💯

**Review by Micaela-Rossi — APPROVED** [review]

**Pull request merged**